crates/stages/benches/criterion.rs

```diff
@@ -0,0 +1,145 @@
+#![allow(dead_code, unused_imports, non_snake_case)]
+
+use criterion::{
+    async_executor::FuturesExecutor, black_box, criterion_group, criterion_main,
+    measurement::WallTime, BenchmarkGroup, Criterion,
+};
+use proptest::{
+    arbitrary::Arbitrary,
+    collection::vec as proptest_vec,
+    prelude::{any_with, ProptestConfig},
+    strategy::{Strategy, ValueTree},
+    test_runner::TestRunner,
+};
+use reth_db::{
+    cursor::{DbCursorRW, DbDupCursorRO, DbDupCursorRW},
+    database::Database,
+    mdbx::{test_utils::create_test_db_with_path, Env, EnvKind, WriteMap},
+    transaction::DbTxMut,
+};
+use reth_primitives::{Header, SealedBlock, TransactionSigned, H256, U256};
+use reth_stages::{
+    stages::TransactionLookupStage, test_utils::TestTransaction, ExecInput, Stage, StageId,
+    StageSetBuilder, UnwindInput,
+};
+use std::{path::Path, sync::Arc, time::Instant};
+criterion_group!(benches, stages);
+criterion_main!(benches);
+
+const VECTORS_FOLDER: &str = "testdata/stages";
+const FILE_PATH: &str = "testdata/stages/blocks";
+const BENCH_DB_PATH: &str = "/tmp/reth-benches-stages";
+const NUM_BLOCKS: usize = 10_000;
+
+pub fn stages(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Stages");
+    group.measurement_time(std::time::Duration::from_millis(2000));
+    group.warm_up_time(std::time::Duration::from_millis(2000));
+
+    let tx = prepare_blocks(NUM_BLOCKS).unwrap();
+
+    measure_txlookup_stage(&mut group, tx);
+}
```

Comment:
i would add manual implementations of various access patterns in a loop, in addition to testing the stage itself. might be easier to compare specific parts if e.g. you have a bench that does X `put`s vs sorted w/ cursor etc.

Reply:
For `append` we know it is fastest if table is empty, question is what is better if we already have some values with random keys in the table and want to insert an additional batch?

Reply:
yeah i should  add those scenarios, however I'll probably put those elsewhere since they're more db specific? this should be only for stage benchmarking? (eg. reading from different tables and aggregating into one; mem vs disk read -> write, etc)

Reply:
Added it on [100cea](https://github.com/paradigmxyz/reth/pull/1130/commits/0100ceac15fdde25bfc6405155f398cad0031f8d). There is a clear difference between presorted insert and unsorted insert. 

```
cargo bench --package reth-db --bench hash_keys
```
<details>

  <summary>Results</summary>

![image](https://user-images.githubusercontent.com/93316087/216758964-6637968b-4275-4106-8003-8798681d4835.png)

</details>

However, it's not that "visible" in the stage itself.  I think because we're allocating all the data into a list before sorting+inserting it. Whereasl we're calling the put on an already existing piece of memory. But will need to run some more benches I guess